app/models/endpoint.rb

```diff
@@ -3,6 +3,7 @@ class Endpoint < ApplicationRecord
 
   default_value_for :verify_ssl, OpenSSL::SSL::VERIFY_PEER
   validates :verify_ssl, :inclusion => {:in => [OpenSSL::SSL::VERIFY_NONE, OpenSSL::SSL::VERIFY_PEER]}
+  validates :port, :numericality => {:only_integer => true, :allow_nil => true, :greater_than => 0}
```

Comment:
I don't believe virtual columns are involved, but...

> it's weird that empty port value or non-numeric value is translated to the DB as 0, isn't it?

I would expect that, as that's how the type casting in Rails works.  See https://robots.thoughtbot.com/how-rails-works-type-casting for decent review.  I can't find how it works in Rails 5, but in rails 4, this is the code that coverts to an integer column: https://github.com/rails/rails/blob/4-1-stable/activerecord/lib/active_record/connection_adapters/column.rb#L169-L178 , and as you can see it's basically just calling .to_i on a String, which `"".to_i # => 0`